how-to-guides/ddp-training/scripts/train_ddp_multiple_runs.py

```diff
@@ -0,0 +1,166 @@
+
+    if rank == 0:
+        # Log params
+        run["parameters"] = params
+
+    print("Start training...")
+    criterion = nn.CrossEntropyLoss()
+    optimizer = optim.SGD(net.parameters(), lr=params["lr"], momentum=params["momentum"])
+    num_of_batches = len(trainloader)
+    for epoch in range(params["epochs"]):
+        trainloader.sampler.set_epoch(epoch)
+        running_loss = 0.0
+        for i, data in enumerate(trainloader, 0):
+            inputs, labels = data
+            images, labels = inputs.cuda(), labels.cuda()
+            optimizer.zero_grad()
+            outputs = net(images)
+            loss = criterion(outputs, labels)
+            loss.backward()
+            optimizer.step()
+            running_loss += loss.item()
+
+        if rank == 0:
+            epoch_loss = running_loss / num_of_batches
+            # Log metrics
+            run["metrics/train/loss"].log(epoch_loss)
+            print(f'[Epoch {epoch + 1}/{params["epochs"]}] loss: {epoch_loss:.3f}')
+
+    print("Finished Training")
+
+
+def test(net, testloader, run, rank):
+
+    correct = 0
+    total = 0
+
+    with torch.no_grad():
+        for data in testloader:
+            images, labels = data
+            images, labels = images.cuda(), labels.cuda()
+            outputs = net(images)
+            _, predicted = torch.max(outputs.data, 1)
+            total += labels.size(0)
+            correct += (predicted == labels).sum().item()
+
+    if rank == 0:
+        acc = 100 * correct // total
+        # Log metrics
+        run["metrics/valid/acc"] = acc
+        print(f"Accuracy of the network on the 10000 test images: {acc} %")
+
+
+def init_distributed():
+
+    # Initializes the distributed backend which will take care of sychronizing nodes/GPUs
+    dist_url = "env://"  # default
+
+    rank = int(os.environ["RANK"])
+    world_size = int(os.environ["WORLD_SIZE"])
+    local_rank = int(os.environ["LOCAL_RANK"])
+
+    dist.init_process_group(backend="nccl", init_method=dist_url, world_size=world_size, rank=rank)
+
+    # this will make all .cuda() calls work properly
+    torch.cuda.set_device(local_rank)
+
+    # synchronizes all the threads to reach this point before moving on
+    dist.barrier()
+
+
+if __name__ == "__main__":
+
+    init_distributed()
+
+    rank = rank = int(os.environ["RANK"])
+    local_rank = int(os.environ["LOCAL_RANK"])
+
+    params = {"batch_size": 256, "lr": 0.001, "epochs": 2, "momentum": 0.9}
+
+    trainloader, testloader = create_data_loader_cifar10(rank=rank, batch_size=params["batch_size"])
+
+    net = torchvision.models.resnet50(weights=None).cuda()
+    net = nn.SyncBatchNorm.convert_sync_batchnorm(net)
+    net = nn.parallel.DistributedDataParallel(net, device_ids=[local_rank])
+
+    # To correctly monitor each GPU usage
+    os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
+    os.environ["CUDA_VISIBLE_DEVICES"] = str(rank)
+
+    # Automatically create and broadcast `custom_run_id` to all processes
```

Comment:
```suggestion
    # (neptune) automatically create and broadcast `custom_run_id` to all processes
```

Reply:
@normandy7  

`(neptune) log...` vs. `(neptune) Log...` vs. `(Neptune) Log...` vs. `(Neptune) log...`

Reply:
`(Neptune) Log...` -- assuming "Log" is what starts a complete sentence. I'd always use sentence case for sentence-like comments.